app/services/reachable_users_service.rb

```diff
@@ -3,24 +3,30 @@ def initialize(campaign)
     @campaign = campaign
     @vaccination_center = campaign.vaccination_center
     @ranking_method = campaign.ranking_method
+    @covering = ::GridCoordsService.new(@vaccination_center.lat, @vaccination_center.lon).get_covering(@campaign.max_distance_in_meters)
   end
 
   def get_users(limit = nil)
     return get_users_with_v2(limit) if @ranking_method == "v2"
     get_users_with_random(limit)
   end
 
+  def get_vaccination_center_grid_query
+    cells = @covering[:cells]
+    "(grid_i, grid_j) IN ((" + cells.map { |sub| sub.join(",") }.join("),(") + "))"
```

Comment:
@poudro pourquoi dans le where tu n'utilise pas un condition sur la distance < distance_max
```
(SQRT( ((:vc_grid_i - u.grid_i) * :grid_cell_size)^2 + ((:vc_grid_j - u.grid_j) * :grid_cell_size)^2 )) < distance_max
```
par exemple ?

Reply:
because in that case you are not using index, but a scan query.

By doing `(grid_i, grid_j) IN ((i,j),...)` you are using the index. 

The covering returned by `GridCoordsService.get_covering` already takes into account the distance to only return possible cells